### PR TITLE
chore(release): 4.0.0

### PR DESCRIPTION
**Pull request by pallabmaiti**
## Description

> Pulled latest from Segment

### CODE REVIEW

File: .github/workflows/build-and-quality-checks.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      - uses: actions/checkout@v3
+      - name: Setup Go ${{ matrix.go }}
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go }}
+
+      - name: Build Go ${{ matrix.go }}
+        run: go build -v ./...
+      - name: Test Go ${{ matrix.go }}
+        run: go test  . -v
```
Comment (lvrach) [inline .github/workflows/build-and-quality-checks.yml:24]: To include tests in sub-directories:

```suggestion
        run: go test  ./... -v
```

Comment (pallabmaiti) [reply .github/workflows/build-and-quality-checks.yml:24]: There are no test cases in subdirectories. Do we need that?

Comment (lvrach) [reply .github/workflows/build-and-quality-checks.yml:24]: future-proofing, if someone adds a sub-package he will not think of updating this yaml file

Comment (pallabmaiti) [reply .github/workflows/build-and-quality-checks.yml:24]: Cool.

**Comment by lvrach** [comment]
@pallabmaiti there some unmerge stuff from main branch

**Review by lvrach — APPROVED** [review]

**Pull request merged**